modules/tinymce/src/plugins/media/test/ts/browser/SubmitTest.ts

```diff
@@ -24,7 +24,7 @@ describe('browser.tinymce.plugins.media.core.SubmitTest', () => {
   const customEmbed =
   '<div style="left: 0px; width: 100%; height: 0px; position: relative; padding-bottom: 56.338%; max-width: 650px;">' +
   '<iframe src="https://www.youtube.com/embed/IcgmSRJHu_8"' +
-  ' width="560" height="314" allowfullscreen="allowfullscreen">' +
+  ' width="560" height="314" allowfullscreen="allowfullscreen" data-mce-fragment="1">' +
```

Comment:
This was removed in the previous PR mentioned in the description and it turns out it wasn't added because this logic wasn't running. I'm still not convinced it should be here, but it was like this in 5.x so it's a separate issue if it shouldn't be here.

Reply:
I ran out of time digging deeper on this but I think it has to do with the attributes being prefixed by a filter in the media plugin. The markFragmentElements will add this attribute to all ast nodes then the unmarkFragmentElements will remove that attribute from all dom elements. However the media plugin will prefix the attributes like this `data-mce-p-data-mce-fragment="1"` then remove that `data-mce-p-` prefix and if I dump the html in the editor body element it still has this prefixed attributes so it's not removing them correctly. I think there is a async media resolver involved here as well so it might be that the element gets converted into an iframe in the dom after the insert is made. So just leaving you with some clues. Might have time to look at this tomorrow unless you solved it by then.

Reply:
I think the filters in the media plugin can be changed to not prefix data attributes. Why it does this is that when a for example placeholder for a media element is used an img retains all the data that an object element would have but since the parser would throw away any invalid attributes they are prefixed this way. So if we either black list the data-mce-fragment or any data-mce- prefixed attribute from being prefixed it should theoretically work.

Reply:
I figured it was probably something like that, so thanks for taking a look 👍 . While out of scope, since it's a quick fix I've added a check so it doesn't copy `data-mce-` prefixed attributes like you suggested. As for cleaning up the rest, I've logged TINY-8479 to look into that as it's a much larger change than I want to make here.